server/server.go

```diff
@@ -231,12 +232,12 @@ func NewServer(ctx context.Context, opts ArgoCDServerOpts) *ArgoCDServer {
 	err = initializeDefaultProject(opts)
 	errors.CheckError(err)
 
-	factory := appinformer.NewSharedInformerFactoryWithOptions(opts.AppClientset, 0, appinformer.WithNamespace(opts.Namespace), appinformer.WithTweakListOptions(func(options *metav1.ListOptions) {}))
+	factory := appinformer.NewSharedInformerFactoryWithOptions(opts.AppClientset, 0, appinformer.WithNamespace(""), appinformer.WithTweakListOptions(func(options *metav1.ListOptions) {}))
 	projInformer := factory.Argoproj().V1alpha1().AppProjects().Informer()
 	projLister := factory.Argoproj().V1alpha1().AppProjects().Lister().AppProjects(opts.Namespace)
 
 	appInformer := factory.Argoproj().V1alpha1().Applications().Informer()
-	appLister := factory.Argoproj().V1alpha1().Applications().Lister().Applications(opts.Namespace)
+	appLister := factory.Argoproj().V1alpha1().Applications().Lister()
```

Comment:
Testing locally I'm getting an RBAC error because the the cluster role doesn't allow access to apps at the cluster level.

Can this behavior be toggled so admins don't have to grant cluster-level app access unless they're using more than one App ns?

Reply:
Same with projects - I think since the factory doesn't have a namespace set, the client is asking for projects at the cluster level, which isn't necessary either in single-namespace or multi-namespace mode.

Reply:
This added to the server cluster role fixed it:

```yaml
- apiGroups:
  - "argoproj.io"
  resources:
  - "applications"
  - "appprojects"
  verbs:
  - list
  - watch
```

Reply:
Good catch! I must admit, I have not tested this yet running from an image in a cluster. And obviously, running it locally has used my kubeadmin user with full cluster level permissions on everything :)

Let me figure out how to also introduce the remote end-to-end test suite into our CI, which would run the end-to-end tests against an image build from the PR and deployed to a real cluster - main concern here is time for the CI to run, though.